coordinator-tool/src/bin/ec.rs

```diff
@@ -0,0 +1,895 @@
+//! A tool for the election coordinator to gather data from the chain, and to
+//! coordinate finalization of the election.
+
+use anyhow::Context;
+use clap::Parser;
+use concordium_governance_committee_election as contract;
+use concordium_rust_sdk::{
+    contract_client::{self, ViewError},
+    indexer,
+    smart_contracts::common::{
+        self as concordium_std, AccountAddress, Amount, OwnedEntrypointName,
+    },
+    types::{
+        hashes::TransactionHash, queries::BlockInfo, AbsoluteBlockHeight, AccountAddressEq,
+        AccountIndex, AccountTransactionEffects, BlockItemSummaryDetails, ContractAddress,
+        WalletAccount,
+    },
+    v2::{self as sdk, BlockIdentifier},
+};
+use concordium_std::schema::SchemaType;
+use contract::GuardiansState;
+use eg::{
+    election_manifest::{ContestIndex, ElectionManifest},
+    election_parameters::ElectionParameters,
+    election_record::PreVotingData,
+    guardian_public_key::GuardianPublicKey,
+    joint_election_public_key::Ciphertext,
+    verifiable_decryption::{
+        DecryptionProofResponseShare, DecryptionShareResult, VerifiableDecryption,
+    },
+};
+use futures::TryStreamExt;
+use indicatif::{ProgressBar, ProgressStyle};
+use std::collections::{BTreeMap, BTreeSet};
+
+/// Command line configuration of the application.
+#[derive(Debug, clap::Parser)]
+#[command(author, version, about)]
+struct Args {
+    /// The node used for querying
+    #[arg(
+        long = "node",
+        help = "The node endpoint.",
+        default_value = "http://localhost:20001",
+        global = true
+    )]
+    node_endpoint: concordium_rust_sdk::v2::Endpoint,
+    #[command(subcommand)]
+    command:       Command,
+}
+
+#[derive(Debug, clap::Subcommand)]
+enum Command {
+    /// For each account compute the average amount of CCD held
+    /// during the period.
+    #[command(name = "initial-weights")]
+    InitialWeights(RangeWithOutput),
+    /// Look for delegations of the vote during the election period.
+    #[command(name = "final-weights")]
+    FinalWeights {
+        #[arg(long = "out", help = "File to output data into.")]
+        out:             Option<std::path::PathBuf>,
```

Comment:
I don't think it's clear what this is? First I expected it to be the output path of the final weights file, but there is a separate argument for that. Can you clarify in the help text?

Reply:
Added. This has the details on who delegated to whom.